### DISCUSSION TITLE
Do all versions of WSA go on Microsoft Store?

### DISCUSSION THREAD

**Discussion by Almighty188** [microsoft/WSA#303] [General]
I have a question, do all versions of WSA go on Microsoft Store as it seems when looking at https://store.rg-adguard.net and ProductId  9P3395VX91NR - it still has 2403 and not 2404 for the fast, slow, Release Preview channels as I thought it was supposed to be the same as the Windows Subsystem for Android™ Preview Program channels.

**Comment by foxhacker0000** [comment]
same

**Reply by Almighty188** [reply]
Both Microsoft Store and https://store.rg-adguard.net/ still has 2303.40000.5.0 for fast/slow/Release Preview and not 2304.40000.3.0.

**Reply by MustardChef** [reply]
> Both Microsoft Store and https://store.rg-adguard.net/ still has 2303.40000.5.0 for fast/slow/Release Preview and not 2304.40000.3.0.

It seems that those (2304.40000.3.0 and 2304.40000.4.0) are released on a seperate channel for the WSA Preview Program (different from Fast (WIF), Slow (WIS) and Retail) and so they will not show up on https://store.rg-adguard.net/.

**Reply by Almighty188** [reply]
Yes, as 2303.40000.2.0 was also a version that was not on  https://store.rg-adguard.net/.

**Comment by foxhacker0000** [comment]
so how we can download it via direct link like https://store.rg-adguard.net/ ???????

i dont want to download it via microsoft store because microsoft store ui have many bugs

**Reply by MustardChef** [reply]
> so how we can download it via direct link like https://store.rg-adguard.net/ ???????
> 
> i dont want to download it via microsoft store because microsoft store ui have many bugs

You will have to join the Windows Preview Program, or wait for someone to post a temp link to the MSIX, or you can get it from here https://github.com/MustardChef/WSAPackages

**Reply by Almighty188** [reply]
> so how we can download it via direct link like https://store.rg-adguard.net/ ???????
> 
> i dont want to download it via microsoft store because microsoft store ui have many bugs

According to this thread, WSA Preview is supposed to download from Microsoft Store.
https://github.com/microsoft/WSA/discussions/200

https://store.rg-adguard.net/ queries Microsoft Store as it seems for the last few months, of all the version(s) that were announced, only three versions not show up on https://store.rg-adguard.net/ which are 2303.40000.2.0, 2304.40000.3.0 and 2304.40000.4.0.  So if the above thread was correct in how WSA Preview is distributed then the issue is those three versions were not sent to Microsoft store.

**Reply by Almighty188** [reply]
> > so how we can download it via direct link like https://store.rg-adguard.net/ ???????
> > i dont want to download it via microsoft store because microsoft store ui have many bugs
> 
> You will have to join the Windows Preview Program, or wait for someone to post a temp link to the MSIX, or you can get it from here https://github.com/MustardChef/WSAPackages

Thanks for providing the archive, now a question since you also use MagiskOnWSALocal, what needs to be done so it will build on MagiskOnWSALocal since for example, the fast channel uses a file named wsa-WIF.zip, is there some way to convert the MSIX to zip so MagiskOnWSALocal can use it?

**Reply by MustardChef** [reply]
> > > so how we can download it via direct link like https://store.rg-adguard.net/ ???????
> > > i dont want to download it via microsoft store because microsoft store ui have many bugs
> > 
> > 
> > You will have to join the Windows Preview Program, or wait for someone to post a temp link to the MSIX, or you can get it from here https://github.com/MustardChef/WSAPackages
> 
> Thanks for providing the archive, now a question since you also use MagiskOnWSALocal, what needs to be done so it will build on MagiskOnWSALocal since for example, the fast channel uses a file named wsa-WIF.zip, is there some way to convert the MSIX to zip so MagiskOnWSALocal can use it?

@Almighty188 
Well, just rename the WSA MSIX file to wsa-WIF.zip and create a folder called `download`. Copy the .zip file to the folder. Now when selecting options for `build.sh` select WIF,  and then specify to skip download of WSA so that it uses the specific .zip that you provide it and then you are free to choose the other options.

Example: 
```shell
sudo git clone https://github.com/MustardChef/MagiskOnWSALocal.git
sudo mkdir ~/MagiskOnWSALocal/download
cd ~/MagiskOnWSALocal/download
sudo cp "/mnt/c/Users/MustardChef/Downloads/MicrosoftCorporationII.WindowsSubsystemForAndroid_2304.40000.4.0_neutral_~_8wekyb3d8bbwe.Msixbundle" wsa-WIF.zip 
cd ~/MagiskOnWSALocal/scripts
./install_deps.sh
sudo ./build.sh --arch x64 --release-type WIF --magisk-ver canary --gapps-brand MindTheGapps --gapps-variant pico --root-sol XXXX --compress-format zip --remove-amazon --skip-download-wsa
```

Hope this helps,
MustardChef

**Reply by Almighty188** [reply]
Thanks, didn't realize renaming it would work as I am not familiar with what format MSIX files are and thought some form of conversions would be needed.  I thought if the file already existed in the download folder, then it won't download it.

**Reply by rp1231** [reply]
> > > > so how we can download it via direct link like https://store.rg-adguard.net/ ???????
> > > > i dont want to download it via microsoft store because microsoft store ui have many bugs
> > > 
> > > 
> > > You will have to join the Windows Preview Program, or wait for someone to post a temp link to the MSIX, or you can get it from here https://github.com/MustardChef/WSAPackages
> > 
> > 
> > Thanks for providing the archive, now a question since you also use MagiskOnWSALocal, what needs to be done so it will build on MagiskOnWSALocal since for example, the fast channel uses a file named wsa-WIF.zip, is there some way to convert the MSIX to zip so MagiskOnWSALocal can use it?
> 
> @Almighty188 Well, just rename the WSA MSIX file to wsa-WIF.zip and create a folder called `download`. Copy the .zip file to the folder. Now when selecting options for `build.sh` select WIF, and then specify to skip download of WSA so that it uses the specific .zip that you provide it and then you are free to choose the other options.
> 
> Example:
> 
> ```shell
> sudo git clone https://github.com/MustardChef/MagiskOnWSALocal.git
> sudo mkdir ~/MagiskOnWSALocal/download
> cd ~/MagiskOnWSALocal/download
> sudo cp "/mnt/c/Users/MustardChef/Downloads/MicrosoftCorporationII.WindowsSubsystemForAndroid_2304.40000.4.0_neutral_~_8wekyb3d8bbwe.Msixbundle" wsa-WIF.zip 
> cd ~/MagiskOnWSALocal/scripts
> ./install_deps.sh
> sudo ./build.sh --arch x64 --release-type WIF --magisk-ver canary --gapps-brand MindTheGapps --gapps-variant pico --root-sol XXXX --compress-format zip --remove-amazon --skip-download-wsa
> ```
> 
> Hope this helps, MustardChef

Hi, I followed all the steps and they completed successfully.
But nothing was produced in the output folder.
I'm using windows subsystem for linux.
Where does the final folder get outputted?
@MustardChef

**Reply by MustardChef** [reply]
> > > > > so how we can download it via direct link like https://store.rg-adguard.net/ ???????
> > > > > i dont want to download it via microsoft store because microsoft store ui have many bugs
> > > > 
> > > > 
> > > > You will have to join the Windows Preview Program, or wait for someone to post a temp link to the MSIX, or you can get it from here https://github.com/MustardChef/WSAPackages
> > > 
> > > 
> > > Thanks for providing the archive, now a question since you also use MagiskOnWSALocal, what needs to be done so it will build on MagiskOnWSALocal since for example, the fast channel uses a file named wsa-WIF.zip, is there some way to convert the MSIX to zip so MagiskOnWSALocal can use it?
> > 
> > 
> > @Almighty188 Well, just rename the WSA MSIX file to wsa-WIF.zip and create a folder called `download`. Copy the .zip file to the folder. Now when selecting options for `build.sh` select WIF, and then specify to skip download of WSA so that it uses the specific .zip that you provide it and then you are free to choose the other options.
> > Example:
> > ```shell
> > sudo git clone https://github.com/MustardChef/MagiskOnWSALocal.git
> > sudo mkdir ~/MagiskOnWSALocal/download
> > cd ~/MagiskOnWSALocal/download
> > sudo cp "/mnt/c/Users/MustardChef/Downloads/MicrosoftCorporationII.WindowsSubsystemForAndroid_2304.40000.4.0_neutral_~_8wekyb3d8bbwe.Msixbundle" wsa-WIF.zip 
> > cd ~/MagiskOnWSALocal/scripts
> > ./install_deps.sh
> > sudo ./build.sh --arch x64 --release-type WIF --magisk-ver canary --gapps-brand MindTheGapps --gapps-variant pico --root-sol XXXX --compress-format zip --remove-amazon --skip-download-wsa
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Hope this helps, MustardChef
> 
> Hi, I followed all the steps and they completed successfully. But nothing was produced in the output folder. I'm using windows subsystem for linux. Where does the final folder get outputted? @MustardChef

- It should be in the `output` folder within the MagiskOnWSALocal folder
- Make sure that you have used
```sudo git clone https://github.com/LSPosed/MagiskOnWSALocal.git``` 
- Make sure that there were no errors during the building process

@rp1231 I highly recommend you to remove the MagiskOnWSALocal Folder and reclone and refollow the steps

**Reply by Almighty188** [reply]
> > > > > so how we can download it via direct link like https://store.rg-adguard.net/ ???????
> > > > > i dont want to download it via microsoft store because microsoft store ui have many bugs
> > > > 
> > > > 
> > > > You will have to join the Windows Preview Program, or wait for someone to post a temp link to the MSIX, or you can get it from here https://github.com/MustardChef/WSAPackages
> > > 
> > > 
> > > Thanks for providing the archive, now a question since you also use MagiskOnWSALocal, what needs to be done so it will build on MagiskOnWSALocal since for example, the fast channel uses a file named wsa-WIF.zip, is there some way to convert the MSIX to zip so MagiskOnWSALocal can use it?
> > 
> > 
> > @Almighty188 Well, just rename the WSA MSIX file to wsa-WIF.zip and create a folder called `download`. Copy the .zip file to the folder. Now when selecting options for `build.sh` select WIF, and then specify to skip download of WSA so that it uses the specific .zip that you provide it and then you are free to choose the other options.
> > Example:
> > ```shell
> > sudo git clone https://github.com/MustardChef/MagiskOnWSALocal.git
> > sudo mkdir ~/MagiskOnWSALocal/download
> > cd ~/MagiskOnWSALocal/download
> > sudo cp "/mnt/c/Users/MustardChef/Downloads/MicrosoftCorporationII.WindowsSubsystemForAndroid_2304.40000.4.0_neutral_~_8wekyb3d8bbwe.Msixbundle" wsa-WIF.zip 
> > cd ~/MagiskOnWSALocal/scripts
> > ./install_deps.sh
> > sudo ./build.sh --arch x64 --release-type WIF --magisk-ver canary --gapps-brand MindTheGapps --gapps-variant pico --root-sol XXXX --compress-format zip --remove-amazon --skip-download-wsa
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Hope this helps, MustardChef
> 
> Hi, I followed all the steps and they completed successfully. But nothing was produced in the output folder. I'm using windows subsystem for linux. Where does the final folder get outputted? @MustardChef

This is how I did it:
On the Windows side, get it from:
Download https://github.com/MustardChef/WSAPackages/releases/download/WSA_2304.40000.4.0/MicrosoftCorporationII.WindowsSubsystemForAndroid_2304.40000.4.0_neutral_._8wekyb3d8bbwe.Msixbundle
Rename the downloaded file to wsa-WIF.zip
Delete everything inside the download folder for MagiskOnWSALocal
Place the wsa-WIF.zip file in the MagiskOnWSALocal download folder

In Linux, from the main MagiMagiskOnWSALocal folder:
git pull
./scripts/install_deps.sh
Build MagiskOnWSALocal as usual and remember to choose the fast channel for WSA

the output folder is output under the main MagiskOnWSALocal

**Reply by rp1231** [reply]
Thanks for the help both.
I'll retry the process.

**Reply by MustardChef** [reply]
> Thanks for the help both. I'll retry the process.

@rp1231 Join the [WSABuilds Discord](https://discord.gg/2thee7zzHZ) if you need further help

**Reply by Almighty188** [reply]
> Thanks for the help both. I'll retry the process.

Forgot to mention, what version of Linux are you using under Windows Subsystem for Linux? 
Use lsb_release -a 
to find out.  Paste your output here.

**Reply by rp1231** [reply]
> > Thanks for the help both. I'll retry the process.
> 
> Forgot to mention, what version of Linux are you using under Windows Subsystem for Linux? Use lsb_release -a to find out. Paste your output here.

This is my output:
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.6 LTS
Release:        20.04
Codename:       focal

**Reply by Almighty188** [reply]
> > > Thanks for the help both. I'll retry the process.
> > 
> > 
> > Forgot to mention, what version of Linux are you using under Windows Subsystem for Linux? Use lsb_release -a to find out. Paste your output here.
> 
> This is my output: No LSB modules are available. Distributor ID: Ubuntu Description: Ubuntu 20.04.6 LTS Release: 20.04 Codename: focal

I went through the same thing earlier since you mention there was no output files as you probably saw, your build failed just like someone else mentioned at https://github.com/LSPosed/MagiskOnWSALocal/issues/535

as your build would have failed as follows:
Create EROFS images
mkfs.erofs 1.0
mkfs.erofs: unrecognized option '--chunksize=4096'
usage: [options] FILE DIRECTORY

Generate erofs image from DIRECTORY to FILE, and [options] are:
 -zX[,Y]   X=compressor (Y=compression level, optional)
 -d#       set output message level to # (maximum 9)
 -x#       set xattr tolerance to # (< 0, disable xattrs; default 2)
 -EX[,...] X=extended options
 -T#       set a fixed UNIX timestamp # to all files
 --help    display this help and exit

Available compressors are: lz4hc, lz4
Build: an error has occurred, exit
``

If you did, apt list erofs-utils 
It will show 1.0-3 as the version even when you try apt reinstall erofs-utils
and with Ubuntu v20.04, 1.0-3 is the latest version unless you manually buuld erofs-utils yourself since it needs v1.4 of erofs-utils as the function chunk-based` first added in 1.4.

So what I had to do is upgrade Ubuntu v20.04 to v22.04 by:
[code]sudo apt update 
sudo apt full-upgrade

You have to restart Ubuntu then:

sudo do-release-upgrade
and then after the update is completed, apt list erofs-utils will show 1.4 which is required to even build MagiskOnWSALocal

**Reply by rp1231** [reply]
Hey Thanks,
I upgraded the ubuntu version but was still not getting the output folder.
But then I tried running the run.sh and it showed in the info initially that it was downloading 2303 but when the script finished, in the end it said 2304. So I have it working now!

**Reply by Almighty188** [reply]
Glad you have it working.  When it was not getting the output folder, did you run install_deps.sh first?  Yes, run.sh seems to get that version 2303 but it doesn't actually download WSA because the needed file is already there and then the output folder would have 2304 in the name of the folder.

**Reply by MustardChef** [reply]
> Glad you have it working. When it was not getting the output folder, did you run install_deps.sh first? Yes, run.sh seems to get that version 2303 but it doesn't actually download WSA because the needed file is already there and then the output folder would have 2304 in the name of the folder.

Seem like 2304.40000.5.0 is now on the Insider channel, so there is no need to download the .msix and rename .etc

**Comment by rp1231** [comment]
Yes I had run install_deps.sh first.
But I don't know maybe I was doing something wrong in the whole process.
I ran it from  WSL.
